### ISSUE TITLE
[css-overflow-5] Discrete versus navigation scroll marker mode

### ISSUE THREAD

**Issue by flackr** [w3c/csswg-drafts#12122]
## Overview
When using scroll markers, there are two known main use cases:
1. A table of contents or set of quick navigation points.
2. A set of independently exclusively accessible content (e.g. a carousel, tabs).
3. If there are other use cases, please comment on this issue so we can consider whether they fit into these two modes or not.

Currently, the focus management and behavior of scroll markers is largely designed around the first use case, and requires the developer to use `interactivity` to inert the non-current content should they need the second use case.

However, it would be nice if the developer could opt into this behavior and the browser could use this opt-in to automatically apply appropriate inferred roles to the content.

## Proposal
Authors creating carousels or tab designs would apply `scroll-marker-type: discrete`, making the scroll marker pseudo-elements a [sequential focus navigation scope owner](https://html.spec.whatwg.org/multipage/interaction.html#focus-navigation-scope-owner) for their originating elements. This means that the originating element (and its descendents) follow the scroll marker in focus order. Since only the :target-current scroll marker is focusable, the other scroll markers and their content are not focusable until a different scroll marker is activated.

When this type is applied, the originating element will be given the "tabpanel" role, and the ::scroll-marker-group and ::scroll-marker pseudo-elements will be given the tablist and tab role respectively.

Otherwise, (e.g. for table of contents use cases) the ::scroll-marker-group and ::scroll-marker pseudo-elements will be given the listbox and option respectively.

## Open questions
* Should this be part of the scroll-marker-group property? E.g. `scroll-marker-group: before discrete;`? It probably doesn't make sense (and we couldn't implicitly apply the tablist role) if this was per scroll marker - so it needs to be associated with the common container of the scroll markers and this mean that we don't have this extra property for all elements just in case they have a scroll marker group.
* Does this have an effect on element scroll markers? Presumably applied to the same element that has `scroll-marker-group: contain` per #10916.
* Should this change keyboard focus behavior? E.g. without specifying this property we would have a list-of-links style focus where every scroll marker is in tab order. This would be a breaking change but the impact is likely minimal for now.
* How would this work when scroll markers are established on ::column pseudos? Or would we skip the sequential focus navigation scope order?
  * In particular, some elements will span across multiple ::column pseudo-elements. Could we make the marker a sequential focus navigation scope order only for the fully contained elements or only elements whose first fragment is in that particular ::column?
* How does this affect the accessibility tree? The expectation of the tabs design pattern is that users are not confronted with content from the inactive tabs. Does this automatically hide the inactive tabs from the AT?

**Comment by aleventhal** [comment]
I think the columns use case might possibly be a third use case. We might decide to expose it differently to the a11y tree than the tabpanel case. Would it be crazy to have scroll-marker-type have 3 values? loose, discrete, and set (modulo bikeshedding). If "set", it would automatically apply CSS columns and any keyboard/axtree improvements we come up with.

Regarding how 'discrete' affects the ax tree, and whether to remove non-active items, I'd be open to it. Would like to hear arguments either way. We also have the option that we simply expose everything but use accessible relations (controls/details/flowto) to enable better AT processing / improve user navigation.

**Comment by flackr** [comment]
@aleventhal @scottaohara @smhigley Based on our discussion, I propose we do following:

Add `discrete | navigation` as a second value to `scroll-marker-group`, default `navigation`. This will affect the roles and behaviours of the associated elements and pseudo-elements as follows:

Navigation:
1. The roles of `::scroll-marker-group` and `::scroll-marker` are `navigation` and `link` respectively.
2. All `::scroll-marker` pseudo-elements are tab stops, in order.
3. Activating a `::scroll-marker` pseudo-element scrolls its originating element into view and sets the [sequential focus navigation starting point](https://html.spec.whatwg.org/#sequential-focus-navigation-starting-point) to the originating element losing focus from the `::scroll-marker`.
4. The roles of the originating elements are unaffected.

Discrete:
1. The roles of `::scroll-marker-group` and `::scroll-marker` are `tablist` and `tab` respectively.
2. Only the active `::scroll-marker` is a tab stop, arrow keys are used to switch between.
3. Activating a `::scroll-marker` pseudo-element scrolls its originating element into view, resulting in the marker being `:target-current` retaining focus on the `::scroll-marker`.
4. Originating elements of `::scroll-marker` are given an implicit role of `tabpanel`. If the originating element is `::column` the `tabpanel` role is given to the originating element of the `::column`.
5. The `::scroll-marker` pseudo-element is a [focus navigation scope owner](https://html.spec.whatwg.org/#focus-navigation-scope-owner) for its associated originating element. This means that tab focus moves from the marker to the content, and since only the active marker is a tab stop per point 2, contents of the inactive tabpanels is not reached unless you switch which tab is active.
6. Content from inactive tabs is also hidden from the AX tree, meaning authors do not need to carefully apply `interactivity: inert` to hide the inactive tabpanels.

The TLDR is that the default `navigation` `::scroll-marker` list behaves like a table of contents list of links, and the `discrete` will automatically have the roles and expected behaviors of the tabs model pattern demonstrated in https://www.w3.org/WAI/ARIA/apg/patterns/carousel/examples/carousel-2-tablist/

**Comment by flackr** [comment]
This is implemented behind experimental web platform features in chromium. I have built a demo to showcase two carousels using each of the two modes at https://flackr.github.io/web-demos/css-overflow/scroll-marker-type/

This is what was proposed in https://github.com/w3c/csswg-drafts/issues/12122#issuecomment-2920507737 using tabs or links instead of discrete or navigation respectively as per resolution in https://github.com/w3c/csswg-drafts/issues/12122#issuecomment-2985032673. Last time we discussed it, the presumption was that we would revisit once we had all of the specifics worked out - I believe they have all worked out as was previously proposed.

Shall we take a resolution to adopt these changes?

Motivation: In my [demo](https://flackr.github.io/web-demos/css-overflow/scroll-marker-type/) I outline all of the automatic benefits of this mode switch that developers no longer need to do extra work to get right. If you contrast this with all of the considerations that must be made with the feature previously, as [outlined here](https://flackr.github.io/web-demos/css-overflow/a11y-examples/), the complexity is significantly reduced making it much more likely that developers using this feature will have the correct behaviors, appropriate roles, and semantics for these use cases.
